src/Compilers/CSharp/Portable/Parser/LanguageParser_Patterns.cs

```diff
@@ -330,14 +335,15 @@ private PatternSyntax ParsePatternContinued(TypeSyntax type, Precedence preceden
                 }
 
                 var positionalPatternClause = _syntaxFactory.PositionalPatternClause(openParenToken, subPatterns, closeParenToken);
-                var result = _syntaxFactory.RecursivePattern(type, positionalPatternClause, propertyPatternClause0, designation0);
+                var result = _syntaxFactory.RecursivePattern(type, positionalPatternClause, lengthPatternClause0, propertyPatternClause0, designation0);
                 return result;
             }
 
-            if (parsePropertyPatternClause(out PropertyPatternClauseSyntax propertyPatternClause))
+            if (parseLengthPatternClause(out LengthPatternClauseSyntax lengthPatternClause) |
```

Comment:
`||`

Reply:
We need both branches to be run. (assigning out params)

Reply:
Ah, makes sense